### PR TITLE
parse: allow `type Foo: Ord` syntactically

### PR DESCRIPTION
**Pull request by Centril**
This addresses:
> (Work still remains to fuse this with free type aliases, but this can be done later.)

in https://github.com/rust-lang/rust/pull/69194.

r? @petrochenkov

### CODE REVIEW

File: src/libsyntax/ast.rs [hunk 1]
```diff
@@ -2524,7 +2524,7 @@ pub enum ItemKind {
     /// A type alias (`type`).
     ///
     /// E.g., `type Foo = Bar<u8>;`.
-    TyAlias(P<Ty>, Generics),
+    TyAlias(Generics, GenericBounds, Option<P<Ty>>),
```
Comment (petrochenkov) [inline src/libsyntax/ast.rs]: Looks like `type A;` is now semantically accepted as a free item.

Comment (Centril) [reply src/libsyntax/ast.rs]: Indeed, but it should be due to foreign types, https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=47b47070b72642e26be4fe693525de8b

Comment (Centril) [reply src/libsyntax/ast.rs]: Oh wait... semantically!... Will fix.

Comment (Centril) [reply src/libsyntax/ast.rs]: Fixed.

**Pull request merged**